--- afsplus_start.c
+++ afsplus_start.c
@@ -72,0 +73 @@
+DECLARESET(LIBS)
@@ -94,3 +95,8 @@
-    if (set_call_funcs(SETNAME(INIT), 1, 1)) {
-        ret = handler(SysBase);
-        set_call_funcs(SETNAME(EXIT), -1, 0);
+#if defined(AFSPLUS_AROS_DIAGNOSTIC_DIRECT_ENTRY)
+    ret = handler(SysBase);
+#else
+    if (set_open_libraries()) {
+        if (set_call_funcs(SETNAME(INIT), 1, 1)) {
+            ret = handler(SysBase);
+            set_call_funcs(SETNAME(EXIT), -1, 0);
+        }
@@ -97,0 +104,2 @@
+    set_close_libraries();
+#endif
